simulations/network_runner.py

```diff
@@ -0,0 +1,288 @@
+        'ZSEQUENCER_BLS_KEY_FILE': '/app/bls_key.json',
+        'ZSEQUENCER_ECDSA_KEY_FILE': '/app/ecdsa_key.json',
+        'ZSEQUENCER_SNAPSHOT_PATH': '/db',
+        'ZSEQUENCER_APPS_FILE': '/app/app.json',
+        'ZSEQUENCER_NODES_FILE': '/app/nodes.json',
+        'ZSEQUENCER_SEQUENCER_SABOTAGE_SIMULATION_TIMESERIES_NODES_STATE_FILE':
+            '/app/sabotage_simulation_timeseries.json'
+    }
+
+    cmd = ["docker", "run", "-d", "--name", container_name]
+    cmd.extend(["--network", DOCKER_NETWORK_NAME])
+
+    for host_path, container_path in volumes.items():
+        cmd.extend(["-v", f"{host_path}:{container_path}"])
+
+    for key, value in docker_env.items():
+        cmd.extend(["-e", f"{key}={value}"])
+
+    port = env_variables.get('ZSEQUENCER_PORT', '6000')
+    cmd.extend(["-p", f"{port}:{port}"])
+    cmd.append(image_name)
+
+    try:
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error starting container {container_name}: {e}")
+        raise
+
+
+def prepare_simulation_files(node_idx: int, keys: Keys):
+    """Prepare node files in the simulation directory."""
+    # Create node directory
+    node_dir = os.path.join(SIMULATION_DATA_DIR, f'node_{node_idx}')
+    if not os.path.exists(node_dir):
+        os.makedirs(node_dir)
+
+    # Save BLS key
+    bls_key_file = os.path.join(node_dir, 'bls_key.json')
+    bls_key_pair = attestation.new_key_pair_from_string(keys.bls_private_key)
+    bls_key_pair.save_to_file(bls_key_file, f'a{node_idx}')
+
+    # Save ECDSA key
+    ecdsa_key_file = os.path.join(node_dir, 'ecdsa_key.json')
+    encrypted_json = Account.encrypt(keys.ecdsa_private_key, f'b{node_idx}')
+    with open(ecdsa_key_file, 'w') as f:
+        json.dump(encrypted_json, f)
+
+    return {
+        'node_dir': node_dir,
+        'bls_key_file': bls_key_file,
+        'ecdsa_key_file': ecdsa_key_file
+    }
+
+
+def get_node_env_variables(node_idx: int, node_dir: str, sequencer_address: str) -> dict:
+    """Generate environment variables for a node."""
+    shared_env = load_env_shared()
+
+    return {
+        **shared_env,
+        "ZSEQUENCER_APPS_FILE": os.path.join(SIMULATION_DATA_DIR, "apps.json"),
+        "ZSEQUENCER_NODES_FILE": os.path.join(SIMULATION_DATA_DIR, "nodes.json"),
+        "ZSEQUENCER_HOST": f"zsequencer-node-{node_idx}",
+        "ZSEQUENCER_PORT": str(BASE_NODE_PORT + node_idx),
+        "ZSEQUENCER_SNAPSHOT_PATH": os.path.join(node_dir, "db"),
+        "ZSEQUENCER_BLS_KEY_FILE": os.path.join(node_dir, "bls_key.json"),
+        "ZSEQUENCER_BLS_KEY_PASSWORD": f"a{node_idx}",
+        "ZSEQUENCER_ECDSA_KEY_FILE": os.path.join(node_dir, "ecdsa_key.json"),
+        "ZSEQUENCER_ECDSA_KEY_PASSWORD": f"b{node_idx}",
+        "ZSEQUENCER_PROXY_PORT": str(BASE_PROXY_PORT + node_idx),
+        "ZSEQUENCER_INIT_SEQUENCER_ID": sequencer_address,
+        "ZSEQUENCER_SEQUENCER_SABOTAGE_SIMULATION_TIMESERIES_NODES_STATE_FILE":
```

Comment:
I would suggest renaming this variable to `‍ZSEQUENCER_SABOTAGE_SIMULATION_TIMESERIES_FILE`